datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py

```diff
@@ -212,4 +212,11 @@ def _make_counters(self, key=None, counter_data=([], ())):  # Key left in for re
 
     @classmethod
     def _no_instance(cls, inst_name):
-        return inst_name.lower() == 'none' or len(inst_name) == 0 or inst_name == '*' or inst_name.lower() == 'all'
+        # type: (Optional[str]) -> bool
+        return (
+            inst_name is None
+            or inst_name.lower() == 'none'
+            or len(inst_name) == 0
+            or inst_name == '*'
+            or inst_name.lower() == 'all'
```

Comment:
unrelated but how come an instance name of `all` is considered no instance?

Reply:
🤷‍♂️ 

Probably a detail of the PDH counters system and our implementation? I think we treat `inst_name = None` as "collect counters from no particular instance", which sounds equivalent to "collect from _all_ instances", hence we support `'all'` , probably just as a shorthand / to prevent silly mistakes from users?